### PR TITLE
Update QueryBuilder docblock to reflect correct type of bound parameter type

### PR DESCRIPTION
**Pull request by fmagnan**
Hello,

thank you very much for your work, it's awesome!
I found a minor error in PHPDOC for method setParameter. Argument "type" can be a string or null but can be also an int if it comes from \PDO::PARAM_*. So I've changed typehint "string" into "mixed".

### CODE REVIEW

File: lib/Doctrine/ORM/QueryBuilder.php [hunk 1]
```diff
@@ -526,7 +526,7 @@ public function getRootEntities()
      *
      * @param string|integer $key   The parameter position or name.
      * @param mixed          $value The parameter value.
-     * @param string|null    $type  PDO::PARAM_* or \Doctrine\DBAL\Types\Type::* constant
+     * @param mixed|null    $type  PDO::PARAM_* or \Doctrine\DBAL\Types\Type::* constant
```
Comment (lcobucci) [inline lib/Doctrine/ORM/QueryBuilder.php]: Can you use `string|integer|null` instead? `mixed` is a bit too broad

**Review by lcobucci — CHANGES_REQUESTED** [review]
@fmagnan thanks for your contribution! Just a small change 😄

**Review by fmagnan — COMMENTED** [review]
done

**Review by lcobucci — APPROVED** [review]

**Comment by lcobucci** [comment]
@fmagnan awesome! 🚢

**Pull request merged**